include/boost/thread/pthread/condition_variable.hpp

```diff
@@ -112,17 +111,18 @@ namespace boost
 #endif
         int cond_res;
         {
+            timespec const& ts = timeout.getTs();
```

Comment:
I find this weird. if getTs() return now a timespec, why not just define a variable as in
```
const timespec ts =  timeout.getTs();
```
?

I wonder if we shouldn't store timespec on pthread systems and intmax_t in windows in internal_timespec_duration. Would this change add a lot of of conditional compilation in detail/timespec .hpp?

In addition it has no sense to name it timespec.hpp if it can be something else.

What do you think?

Reply:
What about platform_clock, platform_duration and platform_timepoint?

The file could be renamed to platform_time.hpp

Reply:
Agreed. It is weird. I'll change it.

I think you're right that we should probably store timespec on pthread and intmax_t on Windows. That would be the most efficient. I don't think it will add too much conditional compilation.

Reply:
And I like the *platform* names instead of *timespec*.